cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/query/IndexUpdater.java

```diff
@@ -199,9 +199,6 @@ private void updateIndex(final String indexName,
                                                             initialArgs,
                                                             indexName,
                                                             rev);
-                if (parameter == null) {
```

Comment:
That probably needs some investigation if that method is doing the "right thing" I think it used to add `null` to the content values if it couldn't figure it out, and then `null` values would be skipped later in the code iirc, @tomblench Does that sound familiar for the method `com.cloudant.sync.internal.query.IndexUpdater#populateDBParameter` ?